klaas/scripts/apply_separation_model.py

```diff
@@ -71,16 +80,29 @@ def main(configuration_path, data_path, model_path, key, chunksize, yes, verbose
         for var in used_source_features
     ]
 
+    if generation_config:
+        needed_features.extend(generation_config['needed_keys'])
+
     df_generator = read_h5py_chunked(
         data_path,
         key=key,
         columns=training_variables + needed_features,
         chunksize=chunksize,
     )
 
+    if generation_config:
+        training_variables.extend(sorted(generation_config['features']))
+
     log.info('Predicting on data...')
     for df_data, start, end in tqdm(df_generator):
 
+        if generation_config:
+            feature_generation(
```

Comment:
I was wondering if it makes more sense to have a script that is doing only the feature generation rather than each training and application script doing it the same way.

Whst do you guys think?